zerver/lib/events.py

```diff
@@ -913,7 +943,18 @@ def post_process_state(user_profile: UserProfile, ret: Dict[str, Any],
     for client.
     '''
     if 'raw_unread_msgs' in ret:
-        ret['unread_msgs'] = aggregate_unread_data(ret['raw_unread_msgs'])
+        if user_profile is not None:
+            # There are no unread messages for guest users.
+            ret['unread_msgs'] = aggregate_unread_data(ret['raw_unread_msgs'])
+        else:
+            ret['unread_msgs'] = {
+                'count': 0,
+                'pms': [],
+                'streams': [],
+                'huddles': [],
+                'mentions': [],
+            }
```

Comment:
Would `aggregate_unread_data` produce these values without the conditional here?  It'd be nice if this function were pure.

Reply:
Done!